modules/caddyhttp/reverseproxy/selectionpolicies_test.go

```diff
@@ -131,6 +131,50 @@ func TestWeightedRoundRobinPolicy(t *testing.T) {
 	}
 }
 
+func TestWeightedRoundRobinPolicyWithZeroWeight(t *testing.T) {
+	pool := testPool()
+	wrrPolicy := WeightedRoundRobinSelection{
+		Weights:     []int{0, 2, 1},
+		totalWeight: 3,
+	}
+	req, _ := http.NewRequest("GET", "/", nil)
+
+	h := wrrPolicy.Select(pool, req, nil)
+	if h != pool[1] {
+		t.Error("Expected first weighted round robin host to be second host in the pool.")
+	}
+
+	h = wrrPolicy.Select(pool, req, nil)
+	if h != pool[2] {
+		t.Error("Expected second weighted round robin host to be third host in the pool.")
+	}
+
+	h = wrrPolicy.Select(pool, req, nil)
+	if h != pool[1] {
+		t.Error("Expected third weighted round robin host to be second host in the pool.")
+	}
```

Comment:
I think this doesn't decisively cover that it would repeat correctly. Maybe it should do like 6 `Select` to make sure four of them are the 2nd and two of them are the 3rd?